ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/CompactorUtil.java

```diff
@@ -565,4 +566,16 @@ public static Map<String, Integer> getPoolConf(HiveConf hiveConf) {
     }
     return poolConf;
   }
+
+  public static String getCompactPoolName(HiveConf conf, org.apache.hadoop.hive.ql.metadata.Table table) 
+      throws Exception {
+    String poolName;
+    Map<String, String> params = table.getParameters();
+    poolName = params == null ? null : params.get(Constants.HIVE_COMPACTOR_WORKER_POOL);
+    if (StringUtils.isBlank(poolName)) {
+      params = CompactorUtil.resolveDatabase(conf, table.getDbName()).getParameters();
```

Comment:
do we have a test when we define the pool at the db level?

Reply:
Added into one of iceberg compaction q-tests.

Reply:
👍